cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -1063,10 +1064,17 @@ impl ManagedEnvironment {
         }
 
         // Fetch the remote branch into sync branch
-        self.floxmeta
+        match self
+            .floxmeta
             .git
             .fetch_ref("dynamicorigin", &format!("+{sync_branch}:{sync_branch}",))
-            .map_err(ManagedEnvironmentError::FetchUpdates)?;
+        {
+            Ok(_) => {},
+            Err(GitRemoteCommandError::RefNotFound(_)) => {
+                warn!("Upstream environment was deleted.")
```

Comment:
```suggestion
                messages::warning("Upstream environment was deleted.")
```

Reply:
well messaging lives in the frontend crate and isnt available here
if we want to warn explicitly and user facing we should split that out and precheck earlier or return a status..

Reply:
Ok, then if you want to show the user do that, and if not make it a `debug` because it doesn't look good to have logs leaking out to the user.